Add completion for packages in update subcommand

Fuco1 opened the pull request:
## Description

This patch adds completion for the update subcommand, that is, when the user types in `composer update <tab>`.

The code depends on python for the json parsing.  I'm not sure if this is appropriate or if there is a fish-native way to parse json data.

## TODOs:
<!-- Just check off what what we know been done so far. We can help you with this stuff. -->
- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

krader1961 commented on share/completions/composer.fish:
```diff
@@ -20,12 +20,26 @@ function __fish_composer_using_command
+function __fish_composer_packages
+    echo "
+import json
+json_data = open('composer.json')
```
Is it safe to assume *composer.json* is in the CWD?

Using python for this is fine since fish has no ability to handle JSON (this might change in the 3.x releases planned for next year).

Fuco1 replied:
The `composer` command only works when you are in the directory with `composer.json`.  So nobody would usually start it from anywhere else *most likely*.  But if they did the code above would probably fail.

I will add some error handling.

Fuco1 replied:
@krader1961 I totally forgot about this :blush: I'm a bit bummed this last week about some stuff, but I'm going to fix this right now so that I won't forget again!

Fuco1 commented:
Well, so I've also added some missing commands and more suggestions for some other commands too.  Maybe I should rename the pull request? We can also split it into more PRs if you want.

krader1961 commented:
Squash merged as commit 3fe561bc. Thx.

Fuco1 commented:
Awesome, thanks! :) Sorry for the delay.
